### PR TITLE
Wallet edit CLI Manager migration

### PR DESCRIPTION
**Pull request by michaelneuder**
**What type of PR is this?**
Refactor/code health

**What does this PR do? Why is it needed?**
This PR addresses the wallet edit functionality and refactors it as part of https://github.com/prysmaticlabs/prysm/issues/9883. It follows up on https://github.com/prysmaticlabs/prysm/pull/10554, https://github.com/prysmaticlabs/prysm/pull/10686, https://github.com/prysmaticlabs/prysm/pull/10824, https://github.com/prysmaticlabs/prysm/pull/10841, https://github.com/prysmaticlabs/prysm/pull/10890 which do similar migration for the accounts list, delete, backup, exit, and import.

1. `cmd/validator/wallet/edit.go`. This is a new file that parses the CLI flags and constructs the CLI manager with the appropriate options. On the CLI manager, it simply calls the `WalletEdit` method with the `context.Context`(no longer a need for the `cli.Context`). The options specific to the import functionality are: `WithWallet` and  `WithKeymangerOpts`.
2. `cmd/validator/wallet/edit_test.go`. This is the new location the test that was in `validator/accounts/wallet_edit_test.go`. As written, this tests are built for the CLI context interface, so if we are moving the logic for handling to the `cmd/validator/wallet` directory, it makes sense to also move the test here. Note that we duplicate a few functions and a struct into this file: `setupWalletAndPasswordsDir`, `testWalletConfig`, and `setupWalletCtx`. These are defined in different tests in the `validator/accounts` directory, and thus aren't importable into this package. One weird aside, these functions are actually also defined in the `cmd/validator/accounts/delete_test.go` too for the same reason (the test depended on them and couldn't import from test files in `validator/acconts`. When I tried to create a shared library for both `accounts/delete_test.go` and `wallet/edit_test.go` I kept running into the following compile error: https://pastebin.com/2n2VB7Ez. I left a TODO in the code to figure this out to remove the duplication.  
3. `validator/accounts/wallet_edit.go`. This is where we see the benefit of this refactor. We chop out all the CLI parsing stuff from the `EditWalletConfigurationCli` function, and make it the `WalletEdit` method on the AccountsCLIManager pointer receiver. 
4. `validator/accounts/cli_manager.go` and `validator/accounts/cli_options.go`. I added a new option that the wallet edit function depends on: `keymanagerOpts *remote.KeymanagerOpts`.

**Which issues(s) does this PR fix?**
This is part of the keymanager refactor defined in https://github.com/prysmaticlabs/prysm/issues/9883.
It is also described in https://github.com/prysmaticlabs/prysm/issues/10339.

### CODE REVIEW

File: cmd/validator/wallet/edit.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/validator/accounts"
+	"github.com/prysmaticlabs/prysm/validator/accounts/userprompt"
+	"github.com/prysmaticlabs/prysm/validator/accounts/wallet"
+	"github.com/prysmaticlabs/prysm/validator/keymanager"
+	"github.com/prysmaticlabs/prysm/validator/keymanager/remote"
+	"github.com/urfave/cli/v2"
+)
+
+func walletEdit(c *cli.Context) error {
```
Comment (james-prysm) [inline cmd/validator/wallet/edit.go:15]: can we call this remote wallet edit or something? wallet is pretty generic if the feature is only really used for the remote keymanager configs, might be kind of confusing in the future. one of the old functions is called edit remote wallet which shows more what it's meant for.

**Review by james-prysm — COMMENTED** [review]
Changes look good and were easy to follow

File: cmd/validator/wallet/edit.go (hunk 1 above)
Comment (michaelneuder) [reply cmd/validator/wallet/edit.go]: renamed to `remoteWalletEdit`! good call, thanks!

**Review by james-prysm — APPROVED** [review]
LGTM

**Pull request merged**